Venmo/src/main/java/com/braintreepayments/api/VenmoClient.java

```diff
@@ -225,7 +235,16 @@ private void startVenmoActivityForResult(
         sharedPrefsWriter.persistVenmoVaultOption(activity, shouldVault);
         if (observer != null) {
             VenmoIntentData intentData = new VenmoIntentData(configuration, venmoProfileId, paymentContextId, braintreeClient.getSessionId(), braintreeClient.getIntegrationType());
-            observer.launch(intentData);
+            if (request.getFallbackToWeb()) {
+                try {
+                    startAppLinkFlow(activity, intentData);
+                } catch (JSONException | BrowserSwitchException exception) {
+                    braintreeClient.sendAnalyticsEvent("pay-with-venmo.app-links.failure");
```

Comment:
Just want to bump [this comment ](https://github.com/braintree/braintree_ios/pull/1193#issuecomment-1961935900)that we'll need to send paymentContextID for Android too.

Reply:
Yep, I added it to the tags confluence page as well as the android ticket for that workstream